docs/nidcpower/class.rst

```diff
@@ -3,38 +3,26 @@
 Session
 =======
 
-.. py:class:: Session(self, resource_name, channels=None, reset=False, options={})
+.. py:class:: Session(self, resource_name, channels=None, reset=False, options={}, independent_channels=True)
 
     
 
-    Creates and returns a new NI-DCPower session to the power supply or SMU
-    specified in **resource name** to be used in all subsequent NI-DCPower
-    method calls. With this method, you can optionally set the initial
-    state of the following session properties:
+    Creates a new NI-DCPower session to the specified instrument(s) and channel(s) and returns a
+    session handle to be used in all subsequent NI-DCPower method calls.
 
-    -  :py:attr:`nidcpower.Session.simulate`
-    -  :py:attr:`nidcpower.Session.driver_setup`
+    Driver initialization will be dispatched to either :py:meth:`nidcpower.Session._initialize_with_channels`
+    (deprecated) or :py:meth:`nidcpower.Session._initialize_with_independent_channels` depending on the value of the
+    independent_channels argument.
 
-    After calling this method, the session will be in the Uncommitted
-    state. Refer to the `Programming
-    States <REPLACE_DRIVER_SPECIFIC_URL_1(programmingstates)>`__ topic for
-    details about specific software states.
+    If independent_channels is True, this method will combine the channels argument (if any)
+    with the resource_name argument then call :py:meth:`nidcpower.Session._initialize_with_independent_channels`. If a
+    channels argument is supplied, resource_name must refer to a single instrument.
 
-    To place the device in a known start-up state when creating a new
-    session, set **reset** to True. This action is equivalent to using
-    the :py:meth:`nidcpower.Session.reset` method immediately after initializing the
-    session.
+    If independent_channels is False, this method will issue a deprecation warning then pass
+    the remaining arguments directly to :py:meth:`nidcpower.Session._initialize_with_channels`.
```

Comment:
lots of implementation details here. Public docs have no business referencing private methods.